ui/pages/setup_secure.tsx

```diff
@@ -80,12 +80,11 @@ const SetupSecure: NextPage = () => {
           },
         })
         .then(({ data }) => {
-          if ("redirect_to" in data) {
-            window.location.href = data.redirect_to as string;
-            return;
-          }
           if (data?.return_to) {
-            window.location.href = data?.return_to;
+            const flowParam = data.return_to.endsWith("/setup_complete")
+              ? ""
+              : `?flow=${data.id}`;
+            window.location.href = `./setup_complete${flowParam}`;
             return;
           }
           window.location.href = "./error";
```

Comment:
I don't think we should redirect to the error page in this case, let's just send them to the `setup_complete` page with the flow and let that page handle it.